ddtrace/tracer.py

```diff
@@ -90,8 +91,16 @@ def configure(self, enabled=None, hostname=None, port=None, sampler=None,
         if enabled is not None:
             self.enabled = enabled
 
-        if hostname is not None or port is not None:
-            self.writer = AgentWriter(hostname or self.DEFAULT_HOSTNAME, port or self.DEFAULT_PORT)
+        processing_pipeline = None
+        if settings is not None:
+                processing_pipeline = settings.get(PP_KEY)
+
+        if hostname is not None or port is not None or processing_pipeline is not None:
```

Comment:
Not something we have to change now, but I think the `configure()` method must be changed in something else. I don't like the idea to initialize again the `AgentWriter`, especially because if you set the `processing_pipeline` and then in another call you change the `hostname`, you end up with the right hostname but wrong pipeline (and viceversa).

Keeping this minor refactoring in another PR so that it doesn't impact this PR.